index.bs

```diff
@@ -5483,6 +5500,377 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+    construct a corresponding [=Credential Parameters=] [=Object=] and add it to |credentialsArray|.
+ 1. Return [=success=] with data containing |credentialsArray|.
+
+## <dfn>Remove Credential</dfn> ## {#sctn-automation-remove-credential}
+
+The [=Remove Credential=] WebDriver [=extension command=] removes a [=Public Key Credential Sources=] stored on a
+[=Virtual Authenticator=]. It is defined as follows:
+
+<figure id="table-removeCredential" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>DELETE</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credentials/{credentialId}`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |authenticator| be the [=Virtual Authenticator=] identified by |authenticatorId|.
+ 1. If |credentialId| does not match any [=Public Key Credential Source=] managed by |authenticator|, return a [=WebDriver error=]
+     with [=WebDriver error code=] [=invalid argument=].
+ 1. Remove the [=Public Key Credential Source=] identified by |credentialId| managed by |authenticator|.
+ 1. Return [=success=].
+
+## <dfn>Remove All Credentials</dfn> ## {#sctn-automation-remove-all-credentials}
+
+The [=Remove All Credentials=] WebDriver [=extension command=] removes all [=Public Key Credential Sources=] stored on a
+[=Virtual Authenticator=]. It is defined as follows:
+
+<figure id="table-removeAllCredentials" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>DELETE</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credentials`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Remove all [=Public Key Credential Sources=] managed by the [=Virtual Authenticator=] identified by |authenticatorId|.
+ 1. Return [=success=].
+
+## <dfn>Set User Verified</dfn> ## {#sctn-automation-set-user-verified}
+
+The [=Set User Verified=] [=extension command=] sets the |isUserVerified| property on the [=Virtual Authenticator=]. It
+is defined as follows:
```

Comment:
Would it make sense to also have a similar command for `isUserPresent`? Maybe combined into a single command like `/session/{session id}/webauthn/authenticator/{authenticatorId}/state` with both `isUserPresent` and `isUserVerified` parameters?

Reply:
I specifically left `isUserPresent` / `isUserConsenting` as a flag set when the authenticator is created that cannot be changed. Otherwise, a client of the API might expect something like this to work:

```javascript
let authenticatorId = AddVirtualAuthenticator({isUserConsenting: false})
navigator.credentials.create(...).then(resolvePromise);
SetIsUserConsenting({authenticatorId, isConsenting: true});
// promise resolves.
```

Supporting a case like this for CTAP2 software authenticators would add complexity for little value for clients of the API. For a client, an authenticator present that does not collect a gesture indicating user consent or the lack of plugged in authenticators should have the same results.